lib/concurrent/atomic/atomic_fixnum.rb

```diff
@@ -74,10 +68,7 @@ def value=(value)
     #
     #   @return [Fixnum] the current value after incrementation
     def increment
-      @mutex.lock
-      @value += 1
-    ensure
-      @mutex.unlock
+      synchronize { @value += 1 }
```

Comment:
My point is was mainly that someone from Ruby land might assume that AtomicFixnum might behave like exactly like a Fixnum that was atomic. This is not the case; they will be sad when their AtomicFixnum maxes out and raises an exception that they were not expecting. To me it just seems like the naming is asking for misinterpretation.